analyses/cell-type-neuroblastoma-04/exploratory-notebooks/singler-results.Rmd

```diff
@@ -0,0 +1,630 @@
+Some sets of marker genes show high expression across multiple cell type categories in particular for closely related cell types (e.g. monocyte, macrophage, myeloid, dendritic, or T-cells and NK-cells), but there do not appear to be any strongly unexpected gene expression patterns.
+
+
+
+### Consensus validation marker genes
+
+This dotplot shows expression of consensus validation marker genes across matching cell types labeled with SingleR.
+
+```{r}
+# Prepare data frame with singler labels to plot
+singler_recoded_df <- celltype_recoded_wide_df |>
+  # we don't consider the NAs here
+  dplyr::filter(singler != "unknown_singler") |>
+  dplyr::select(cell_id, label_recoded = singler)
+
+
+# get total number of cells per final annotation group and set up y_label
+total_cells_df <- singler_recoded_df |>
+  dplyr::count(label_recoded, name = "total_cells") |>
+  dplyr::arrange(desc(total_cells)) |>
+  dplyr::mutate(y_label = glue::glue("{label_recoded} ({total_cells})"))
+
+singler_order <- total_cells_df$y_label
+total_cells_df$y_label <- factor(total_cells_df$y_label, levels = singler_order)
+```
+
+
+
+```{r}
+# prepare markers for dotplot
+dotplot_consensus_markers_df <- consensus_markers_df |>
+  # consider only matching labels
+  dplyr::filter(validation_group_annotation %in% singler_recoded_df$label_recoded) |>
+  dplyr::select(
+    marker_gene_label = validation_group_annotation,
+    ensembl_gene_id,
+    gene_symbol
+  )
+
+# get the bar order
+consensus_bar_order <- total_cells_df |>
+  dplyr::select(label_recoded, y_label) |>
+  # inner!!!!
+  dplyr::inner_join(
+    dotplot_consensus_markers_df, 
+    by = c("label_recoded" = "marker_gene_label")
+  ) |>
+  dplyr::arrange(y_label) |>
+  dplyr::pull(label_recoded) |>
+  unique()
+```
+
+
+```{r, fig.width = 26, fig.height = 12}
+generate_dotplot(
+  merged_sce = merged_sce,
+  markers_df = dotplot_consensus_markers_df,
+  singler_df = singler_recoded_df,
+  total_cells_df = total_cells_df,
+  expressed_genes = expressed_genes,
+  bar_order = consensus_bar_order, 
+  celltype_palette = recoded_celltype_pal, 
+  min_cells = 0
+)
+```
+
+Again, marker gene expression tends to be high in the matching cell type categories, with the possible exception of `B cell` which has somewhat lower gene expression of fewer genes; but, there is still some expected expression there.
+We also see some "promiscuity" where marker genes for closely-related cell types show high expression. 
+
+
+## Compare expression when SingleR and consensus annotations match
+
+Next, we look at expression of consensus validation marker genes for each normal/harmonized cell type label.
+We compare expression across groups of cells:
+
+* `matching annotations`: Cells whose consensus annotation matches the SingleR annotation
+* `different annotations`: Cells whose consensus annotation differs from the SingleR annotation
+* `consensus not classified`: Cells where the consensus annotation is unknown
```

Comment:
Sorry for the confusion with my initial comment, but I am specifically interested in the gene expression for cells that are classified as a normal cell type in SingleR and classified as unknown in consensus, so I would define these groups differently. I would divide like so: 
- cells that have a non-NE label in SingleR + classified as unknown by consensus
- all other cells that are either NE or not classified by SingleR + classified as unknown by consensus

Basically there are a group of cells for every cell type that `SingleR` classifies but consensus does not and I want to see if they do in fact have expression of the marker genes for that cell type. I would also use the NBAtlas marker genes here.